fenix/app/src/main/java/org/mozilla/fenix/debugsettings/tabs/TabManager.kt

```diff
@@ -0,0 +1,226 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.ContentState
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.observeAsState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Divider
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.ext.maxActiveTime
+import org.mozilla.fenix.tabstray.ext.isNormalTabInactive
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Tab manager UI for [DebugDrawer] that displays the tab counts and allows easy  bulk opening of tabs.
+ */
+@Composable
+fun TabManager(
+    store: BrowserStore,
+    inactiveTabsEnabled: Boolean,
+) {
+    val tabs by store.observeAsState(initialValue = emptyList()) { state -> state.tabs }
+    val privateTabCount = remember(tabs) { tabs.filter { it.content.private }.size }
+    val inactiveTabCount = remember(tabs) {
+        if (inactiveTabsEnabled) {
+            tabs.filter { it.isNormalTabInactive(maxActiveTime) }.size
+        } else {
+            0
+        }
+    }
+    val normalTabCount = remember(tabs) { tabs.size - privateTabCount - inactiveTabCount }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(all = 16.dp),
+    ) {
+        TabCounter(
+            normalTabCount = normalTabCount,
+            inactiveTabCount = inactiveTabCount,
+            privateTabCount = privateTabCount,
+            totalTabCount = tabs.size,
+            inactiveTabsEnabled = inactiveTabsEnabled,
+        )
+    }
+}
+
+@Composable
+private fun TabCounter(
+    normalTabCount: Int,
+    inactiveTabCount: Int,
+    privateTabCount: Int,
+    totalTabCount: Int,
+    inactiveTabsEnabled: Boolean,
+) {
+    Column {
+        Text(
+            text = stringResource(R.string.debug_drawer_tab_manager_tab_count_title),
+            color = FirefoxTheme.colors.textPrimary,
+            style = FirefoxTheme.typography.headline5,
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        TabCountRow(
+            text = stringResource(R.string.debug_drawer_tab_manager_tab_count_normal),
+            count = normalTabCount.toString(),
+        )
+
+        TabCountRow(
+            text = stringResource(R.string.debug_drawer_tab_manager_tab_count_inactive),
+            count = if (inactiveTabsEnabled) {
+                inactiveTabCount.toString()
+            } else {
+                "0*"
```

Comment:
could we just not show the `inactive` tabs count if it's disabled? or replace '0*' with something like `N/A`?

Reply:
It's certainly an option. My thought process was that it'll be more clear to QA, or also really anyone using this feature for the first few times, that the tab counter will _always_ include the inactive tab line item, so it isn't hidden and they'd have to check secret settings to see if inactive tabs are even enabled. 

That being said, this was certainly more of a catch for when the copy was "Normal" versus "Active" tabs. Now that the copy has changed, I'll make this next iteration show/hide the entire inactive section and we can see what that looks like.

Reply:
Update: screenshots have this change